botorch/utils/sample_pareto.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Some methods for sampling the Pareto optimal points. This relies on the pymoo
+package for solving multi-objective optimization problems using genetic algorithms.
+
+TODO: The Pareto solver relies on pymoo 0.6.0, it might be advantageous to consider
+    alternative approaches for multi-objective optimization such as multiple
+    gradient descent algorithms.
+"""
+
+from __future__ import annotations
+
+from math import ceil
+from typing import Optional, Tuple
+
+import torch
+from botorch.exceptions.errors import UnsupportedError
+from botorch.models.deterministic import GenericDeterministicModel
+from botorch.models.model import Model
+from botorch.utils.gp_sampling import get_gp_samples
+from botorch.utils.multi_objective import is_non_dominated
+from botorch.utils.multi_objective.box_decompositions.non_dominated import (
+    FastNondominatedPartitioning,
+)
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.core.problem import Problem
+from pymoo.operators.crossover.sbx import SBX
+from pymoo.operators.mutation.pm import PolynomialMutation
+from pymoo.operators.sampling.rnd import FloatRandomSampling
+from pymoo.optimize import minimize
+from pymoo.termination import get_termination
+from torch import Tensor
+
+
+def pareto_solver(
+    model: GenericDeterministicModel,
+    bounds: Tensor,
+    num_objectives: int,
+    num_generations: int = 100,
+    pop_size: int = 100,
+    num_offsprings: int = 10,
```

Comment:
Hmm are these good defaults? 10 offsprings seems quite small. Why not use the pymoo defaults?